### PR TITLE
[CAZB-7700] Apply decommission date to matrix view

### PR DESCRIPTION
**Pull request by wojw5**
https://informedsolutions.atlassian.net/browse/CAZB-7700

## Description
- applied end_date to dated displayed on matrix
- hidding next 7 days tab when nothing to display
- displaying note conditionally

Screenshot 2022-09-12 at 12 58 02
Screenshot 2022-09-12 at 13 09 57
Screenshot 2022-09-12 at 13 09 37

### CODE REVIEW

**Review by likeuwill — APPROVED** [review]

File: app/services/payments/payment_dates.rb [hunk 1]
```diff
@@ -67,33 +65,56 @@ def parse(date)
...
       {
-        past: (Time.zone.today - 6.days..Time.zone.yesterday).map { |date| parse(date) },
-        next: (Time.zone.today..(Time.zone.today + 6.days)).map { |date| parse(date) }
+        past: ((today - 6.days)..yesterday).map { |date| parse(date) },
+        next: (today..(today + 6.days)).map { |date| parse(date) }
       }
     end
+
+    def today
+      @today ||= Time.zone.today
```
[the remark this answers is not in the document]
Comment (wojw5) [reply app/services/payments/payment_dates.rb:113]: I found it easier to read through the code when reading just `today` and just create this delegation method with additional memoization. Do you think it is the wrong pattern? @likeuwill

[the remark this answers is not in the document]
Comment (likeuwill) [reply app/services/payments/payment_dates.rb:119]: ohh sorry, my bad, i thought u took `ActiveSupport` helper, nmv

File: app/services/payments/payment_dates.rb [hunk 2]
```diff
@@ -18,28 +18,22 @@ class PaymentDates < BaseService
     # ==== Attributes
     # * +charge_start_date+ - date when CAZ started charging.
```
Comment (likeuwill) [inline app/services/payments/payment_dates.rb:19]: missing attributes

File: app/services/payments/payment_dates.rb [hunk 3]
```diff
@@ -52,10 +46,14 @@ def d_day_content_tab
       end
     end
```
Comment (likeuwill) [inline app/services/payments/payment_dates.rb:48]: comment missng

**Review by likeuwill — APPROVED** [review]

File: app/services/payments/payment_dates.rb [hunk 4]
```diff
@@ -67,33 +65,63 @@ def parse(date)
       }
     end
 
+
+
```
Comment (likeuwill) [inline app/services/payments/payment_dates.rb:69]: to many empty lines

**Review by likeuwill — CHANGES_REQUESTED** [review]

**Review by likeuwill — APPROVED** [review]

**Pull request merged**